### ISSUE TITLE
Couldn't load plugin 'LDMP' due to an error when calling its classFactory() method 

### ISSUE THREAD

**Issue by kamaulindhardt** [ConservationInternational/trends.earth#492]
When I try installing the plugin trends.earth I get this following error.  

`Couldn't load plugin 'LDMP' due to an error when calling its classFactory() method `

I have tried reading up on possible ways of tackling the problem but without luck. I am running macOS but I experience the xact same error when running in my parallel windows OS.

Any help from the developers, please? 

```
AttributeError: module 'time' has no attribute 'clock' 
Traceback (most recent call last):
  File "C:\PROGRA~1/QGIS32~1.0/apps/qgis/./python\qgis\utils.py", line 335, in _startPlugin
    plugins[packageName] = package.classFactory(iface)
  File "C:\Users/kamaulindhardt/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\LDMP\__init__.py", line 49, in classFactory
    from LDMP.plugin import LDMPPlugin
  File "C:\PROGRA~1/QGIS32~1.0/apps/qgis/./python\qgis\utils.py", line 799, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "C:\Users/kamaulindhardt/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\LDMP\plugin.py", line 26, in 
    from LDMP.jobs import DlgJobs
  File "C:\PROGRA~1/QGIS32~1.0/apps/qgis/./python\qgis\utils.py", line 799, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "C:\Users/kamaulindhardt/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\LDMP\jobs.py", line 39, in 
    from LDMP.plot import DlgPlotTimeries
  File "C:\PROGRA~1/QGIS32~1.0/apps/qgis/./python\qgis\utils.py", line 799, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "C:\Users/kamaulindhardt/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\LDMP\plot.py", line 18, in 
    import pyqtgraph as pg
  File "C:\PROGRA~1/QGIS32~1.0/apps/qgis/./python\qgis\utils.py", line 799, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "C:\Users\kamaulindhardt\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs\pyqtgraph\__init__.py", line 204, in 
    from .graphicsItems.VTickGroup import *
  File "C:\PROGRA~1/QGIS32~1.0/apps/qgis/./python\qgis\utils.py", line 799, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "C:\Users\kamaulindhardt\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs\pyqtgraph\graphicsItems\VTickGroup.py", line 7, in 
    from .. import functions as fn
  File "C:\PROGRA~1/QGIS32~1.0/apps/qgis/./python\qgis\utils.py", line 799, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "C:\Users\kamaulindhardt\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs\pyqtgraph\functions.py", line 17, in 
    from . import debug
  File "C:\PROGRA~1/QGIS32~1.0/apps/qgis/./python\qgis\utils.py", line 799, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "C:\Users\kamaulindhardt\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs\pyqtgraph\debug.py", line 11, in 
    from . import ptime
  File "C:\PROGRA~1/QGIS32~1.0/apps/qgis/./python\qgis\utils.py", line 799, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "C:\Users\kamaulindhardt\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs\pyqtgraph\ptime.py", line 24, in 
    cstart = systime.clock()  ### Required to start the clock in windows
AttributeError: module 'time' has no attribute 'clock'


Python version: 3.9.5 (tags/v3.9.5:0a7dcbd, May  3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)] 
QGIS version: 3.20.0-Odense Odense, decaadbb31 

Python Path:
C:/PROGRA~1/QGIS32~1.0/apps/qgis/./python
C:\Users\kamaulindhardt\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs
C:\Users\kamaulindhardt\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs
C:\Users\kamaulindhardt\AppData\Roaming\QGIS\QGIS3\profiles\default\python\plugins\LDMP\ext-libs

```

Because I run macOS I also tried there. Here I get the same error..

`Couldn't load plugin 'LDMP' due to an error when calling its classFactory() method `

```
KeyError: 'Can only index numba types with slices with no start or stop, got 4.' 
Traceback (most recent call last):
  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 335, in _startPlugin
    plugins[packageName] = package.classFactory(iface)
  File "/Users/kamaulindhardt_1/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/LDMP/__init__.py", line 49, in classFactory
    from LDMP.plugin import LDMPPlugin
  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 799, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "/Users/kamaulindhardt_1/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/LDMP/plugin.py", line 24, in 
    from LDMP.download_data import DlgDownload
  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 799, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "/Users/kamaulindhardt_1/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/LDMP/download_data.py", line 30, in 
    from LDMP.calculate import DlgCalculateBase, get_script_slug
  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 799, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "/Users/kamaulindhardt_1/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/LDMP/calculate.py", line 59, in 
    from LDMP.calculate_numba import *
  File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 799, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "/Users/kamaulindhardt_1/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/LDMP/calculate_numba.py", line 130, in 
    @cc.export('ldn_total_deg', 'f4[4](i2[:,:], b1[:,:], f4[:,:])')
  File "/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/numba-0.50.1-py3.8-macosx-10.13.0-x86_64.egg/numba/pycc/cc.py", line 136, in export
    fn_args, fn_retty = sigutils.normalize_signature(sig)
  File "/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/numba-0.50.1-py3.8-macosx-10.13.0-x86_64.egg/numba/core/sigutils.py", line 24, in normalize_signature
    parsed = _parse_signature_string(sig)
  File "/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/numba-0.50.1-py3.8-macosx-10.13.0-x86_64.egg/numba/core/sigutils.py", line 14, in _parse_signature_string
    return eval(signature_str, {}, types.__dict__)
  File "", line 1, in 
  File "/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/numba-0.50.1-py3.8-macosx-10.13.0-x86_64.egg/numba/core/types/abstract.py", line 185, in __getitem__
    ndim, layout = self._determine_array_spec(args)
  File "/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/numba-0.50.1-py3.8-macosx-10.13.0-x86_64.egg/numba/core/types/abstract.py", line 210, in _determine_array_spec
    raise KeyError(f"Can only index numba types with slices with no start or stop, got {args}.")
KeyError: 'Can only index numba types with slices with no start or stop, got 4.'


Python version: 3.8.7 (default, May 24 2021, 13:02:34) [Clang 12.0.5 (clang-1205.0.22.9)] 
QGIS version: 3.20.0-Odense Odense, decaadbb31 

Python Path:
/Applications/QGIS.app/Contents/MacOS/../Resources/python
/Users/kamaulindhardt_1/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/LDMP/ext-libs
/Users/kamaulindhardt_1/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/LDMP/ext-libs
/Users/kamaulindhardt_1/Library/Application Support/QGIS/QGIS3/profiles/default/python
/Users/kamaulindhardt_1/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins
/Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins
/Applications/QGIS.app/Contents/MacOS/lib/python38.zip
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/pyproj-2.6.0-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/matplotlib-3.3.0-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/numpy-1.20.1-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/netCDF4-1.5.4-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/patsy-0.5.1-py3.8.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/numba-0.50.1-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/snuggs-1.4.7-py3.8.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/pandas-1.1.0-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/GDAL-3.2.3-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/affine-2.3.0-py3.8.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/cftime-1.2.1-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/Fiona-1.8.13.post1-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/opencv_contrib_python-4.3.0.36-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/Pillow-7.2.0-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/lib-dynload
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/geopandas-0.8.1-py3.8.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/scipy-1.5.1-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/Rtree-0.9.4-py3.8.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/click_plugins-1.1.1-py3.8.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/rasterio-1.1.5-py3.8-macosx-10.13.0-x86_64.egg
/Applications/QGIS.app/Contents/MacOS/lib/python3.8/site-packages/statsmodels-0.11.1-py3.8-macosx-10.13.0-x86_64.egg
/Users/kamaulindhardt_1/Library/Application Support/QGIS/QGIS3/profiles/default/python
/Users/kamaulindhardt_1/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins
/Users/kamaulindhardt_1/Library/Mobile Documents/com~apple~CloudDocs/MSc Wageningen University /Internship ICRAF-ERA/QGIS
/Users/kamaulindhardt_1/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/qgis2web
/Users/kamaulindhardt_1/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins
```

**Comment by azvoleff** [comment]
Which plugin version are you using? The Mac OS error you reference looks like #318 (now fixed in develop but not yet released).

The first looks to be an error in pyqtgraph related to Python 3.9 (now used on QGIS 3.20). Looks to now be fixed: pyqtgraph/pyqtgraph#980.

Will update pyqtgraph for the upcoming (within next few days) 1.0.6 release.

**Comment by azvoleff** [comment]
Fixed in 1.0.6
